enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,1050 @@
+  * Refactor / modernize the operator (it's quite old one):
+    * Check and remove manifests/03_configmap.yaml, it does not seem to be useful.
+    * Check and remove manifests/03_service.yaml, it does not seem to be useful (at least now).
+    * Use DeploymentController from library-go to sync Deployments.
+    * Get rid of `common/` package? It does not seem to be useful.
+    * Use StaticResourceController for static content.
+    * Move volume snapshot CRD creation from the operator to CVO.
+      * The CRDs are static, and they were created by the operator only to block upgrade from a version when user could
+        manually install v1alpha1 version of them. The CRDs are v1 for few releases already.
+    * Move creation of manifests/08_webhook_service.yaml from CVO to the operator - it needs to be created in the
+      management cluster.
+    * Fix metrics collection - currently it looks like we don't collect neither operator nor operand metrics even
+      in standalone OCP.
+  * Add a new cmdline option for the guest cluster kubeconfig file location.
+    * Parse both kubeconfigs:
+      * One from projected service account, which leads to the management cluster.
+      * Second from the new cmdline option introduced above. This one leads to the guest cluster.
+  * Only on HyperShift:
+    * Replace namespaces in all Deployments, PDBs and other objects that are created in the management cluster. They
+      must be created in the same namespace as the operator.
+    * Update ValidationWebhookConfiguration to point directly to URL exposed by manifests/08_webhook_service.yaml
+      instead of a Service. The Service is not available in the guest cluster.
+    * When interacting with Kubernetes API, carefully choose the right kubeconfig to watch / create / update objects in
+      the right cluster.
+    * Pass only the guest kubeconfig to the operands (both the webhook and csi-snapshot-controller).
+    * Allow DaemonSet with zero replicas to handle cluster without any worker nodes.
+    * Phase 2: create ServiceAccount + RBACs in the guest cluster for the operand.
+  
+* **csi-snapshot-controller and the webhook**:
+  * No changes there! They will run in the management cluster, but they get kubeconfig to the guest cluster. 
+
+#### cluster-storage-operator
+
+**AWS EBS CSI driver deployment**:
+```mermaid
+flowchart TB
+
+subgraph management-cluster
+    control-plane-operator -->|create| cluster-storage-operator[cluster-storage-operator Deployment]
+    cluster-storage-operator -->|create| aws-ebs-csi-driver-operator[aws-ebs-csi-driver-operator Deployment]
+    aws-ebs-csi-driver-operator -->|create| aws-ebs-csi-driver-controller[aws-ebs-csi-driver-controller Deployment]
+end
+
+subgraph guest-cluster
+    aws-ebs-csi-driver-operator -->|create| aws-ebs-csi-driver-node[[aws-ebs-csi-driver-node DaemonSet]]
+end
+```
+
+**cluster-storage-operator and aws-ebs-csi-driver-operator runtime objects**:
```

Comment:
The same comment about combining the diagrams applies here. Both cluster-storage-operator and aws-ebs-csi-driver-operator are reconciling resources in both clusters, but it isn't clear from the way the diagrams are drawn.

What does aws-ebs-csi-driver-controller do? It isn't linked to any resources in these diagrams.

Reply:
aws-ebs-csi-driver-controller is the CSI driver itself, I added some arrows to it as it creates PVs for PVCs and so on.

I added a combined picture below, but again, I am hitting mermaid limits.